Merge up 3.9.x into 4.0.x

sdelamo opened the pull request.

kgeri committed:
Support Range requests when returning SystemFiles (#8553)

Fixes #7911

alvarosanchez committed:
Refresh Maven core plugin versions (#8623)

micronaut-build committed:
Bump micronaut-security to 3.9.1 (#8628)

micronaut-build committed:
Bump micronaut-views to 3.8.1 (#8629)

micronaut-build committed:
Bump micronaut-spring to 4.5.0 (#8615)

micronaut-build committed:
Bump micronaut-servlet to 3.3.4 (#8634)

sdelamo committed:
Merge branch '3.6.x' into 3.7.x

sdelamo committed:
Merge branch '3.7.x' into 3.8.x

auke- committed:
fix: handle inner classes in bean import (#8566) (#8567)

timyates committed:
Disable prediction and distribution for test-suite-kotlin (#8612)

* Disable predition and distribution for test-suite-kotlin

* Trigger CI again

micronaut-build committed:
build: Bump micronaut-servlet to 3.3.5 (#8640)

sdelamo committed:
doc: extract cors docs to sections (#8637)

sdelamo committed:
Merge branch '3.6.x' into 3.7.x

sdelamo committed:
Merge branch '3.7.x' into 3.8.x

sdelamo committed:
Merge branch '3.8.x' into 3.9.x

sdelamo committed:
build: Micronaut Micrometer 4.7.1

sdelamo committed:
build: Security 3.9.2

micronaut-build committed:
[skip ci] Release v3.8.2

micronaut-build committed:
Back to 3.8.3-SNAPSHOT

sdelamo committed:
Merge branch '3.8.x' into 3.9.x

msupic committed:
Fixed property placeholder expression issue (issue #8633) (PR #8645)

juanpabloprado committed:
chore(doc): upgrade lombok patch version (#8648)

micronaut-build committed:
Bump micronaut-oracle-cloud to 2.3.2 (#8658)

sdelamo committed:
ci: sync as in project template (#8671)

mattmoss committed:
fix: Handle WebLogic classloader URIs (#8646)

timyates committed:
bug: fix Cors for alternate local addresses (#8642)

timyates committed:
Add discovery 3.3.0 BOM to build [skip ci] (#8659)

Once 3.3.0 is released with a BOM, we can add it to the catalog

micronaut-build committed:
[skip ci] Release v3.5.6

micronaut-build committed:
Back to 3.5.7-SNAPSHOT

sdelamo committed:
Merge branch '3.5.x' into 3.6.x

micronaut-build committed:
[skip ci] Release v3.6.5

micronaut-build committed:
Back to 3.6.6-SNAPSHOT

sdelamo committed:
Merge branch '3.6.x' into 3.7.x

micronaut-build committed:
[skip ci] Release v3.7.6

micronaut-build committed:
Back to 3.7.7-SNAPSHOT

sdelamo committed:
Merge branch '3.7.x' into 3.8.x

sdelamo committed:
build: micronau-tracing 4.5.0 (#8672)

Close #8548

micronaut-build committed:
build: bump micronaut-openapi to 4.8.2 (#8656)

micronaut-build committed:
[skip ci] Release v3.8.3

micronaut-build committed:
Back to 3.8.4-SNAPSHOT

sdelamo committed:
Merge branch '3.8.x' into 3.9.x

micronaut-build committed:
Bump micronaut-micrometer to 4.7.2 (#8681)

micronaut-build committed:
build: Bump micronaut-azure to 3.7.1 (#8682)

micronaut-build committed:
Bump micronaut-azure to 3.8.0 (#8683)

sdelamo committed:
build: back to managed-micrometer 1.9.4

changed accidentally via https://github.com/micronaut-projects/micronaut-core/pull/8681

timyates committed:
Fix build by re-adding the managed-micrometer version (#8685)

This PR https://github.com/micronaut-projects/micronaut-core/pull/8681 deleted the micrometer version

Here we replace it with the version of Micrometer that is found in micronaut-micrometer 4.7.2

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

yawkat committed:
Add docs on self-signed cert setup (#8684)

micronaut-build committed:
Bump micronaut-aws to 3.13.1 (#8694)

micronaut-build committed:
Bump micronaut-tracing to 4.4.1 (#8692)

sdelamo committed:
Merge branch '3.7.x' into 3.8.x

sdelamo committed:
Merge branch '3.8.x' into 3.9.x

sdelamo committed:
Revert "build: micronau-tracing 4.5.0 (#8672)"

This reverts commit 8ab03d2c5ea2c7e3a92d7830256829301c117f05.

sdelamo committed:
Merge branch '3.9.x' into 4.0.x

sdelamo committed:
test: require docker for test (#8698)

dstepanov commented on core/src/main/java/io/micronaut/core/exceptions/UncheckedIOException.java:
```diff
@@ -0,0 +1,29 @@
+ * @author Sergio del Amo
+ * @since 3.5.6
+ */
+public class UncheckedIOException extends RuntimeException {
```
Can we use https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html

sdelamo replied:
resolved via https://github.com/micronaut-projects/micronaut-core/pull/8701/commits/1ab8748fba370bf20c03819d8e2085fbcbed1d93

sdelamo committed:
build: add back managed-micronaut-discovery (#8702)

* build: Micronaut Build Plugin 5.4.2

* build: add back managed-micronaut-discovery

otherwise ./gradlew bom:checkVersionCatalogueCompatibility fails

it was removed in https://github.com/micronaut-projects/micronaut-core/pull/8659

sdelamo committed:
build: micronau-tracing 4.5.0 (#8700)

* Silence Micronaut Tracing BOM validation problems

Co-authored-by: Cedric Champeau <cedric.champeau@gmail.com>

sdelamo committed:
Merge branch '3.8.x' into 3.9.x

sdelamo committed:
Merge branch '3.9.x' into mergeup

sdelamo committed:
remove unused file
